src/client/Microsoft.Identity.Client/Platforms/net5win/Net5WamInterop.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Security.Authentication.Web.Core;
+using Windows.Security.Credentials;
+using Windows.UI.ApplicationSettings;
+using WinRT;
+
+namespace Microsoft.Identity.Client.Platforms.net5win
+{
+    internal static class WebAuthenticationCoreManagerInterop
+    {
+        public static IAsyncOperation<WebTokenRequestResult> RequestTokenForWindowAsync(IntPtr hWnd, WebTokenRequest request)
```

Comment:
>RequestTokenForWindowAsync [](start = 61, length = 26)

is this because it doesn't work w/a console? what does "window" refer to?

Reply:
Window here refers to a window handle, i.e. a pointer to a "window" object. "Window" as in graphical element. In WinForms this is a top level Form, in WPF it's a Page. 

Win32 APIs have extensive support for "Windows", e.g. [GetForegroundWindow](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getforegroundwindow) will get you a handle (pointer) to whatever program is focused (most likely your browser since you are reading this). 

We have access to WAM APIs via WinRT and via COM. WinRT which was originally designed for UWP apps. In UWP apps, you don't need to specify a window handle, because the underlying logic knows how to get it. So in order for non-UWP apps (call them win32 apps) to call into WAM, we need different APIs, which take in a window handle obj, and this interop logic is here to add these APIs, which are still accessible via COM.

Reply:
cool. thank you.

---
In reply to: [537350869](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2279#discussion_r537350869) [](ancestors = 537350869)